edb/lib/std/25-numoperators.eql

```diff
@@ -934,19 +1038,29 @@ std::`%` (n: std::decimal, d: std::decimal) -> std::decimal
 };
 
 
+# need an explicit operator for int64 in order to guarantee the result
+# is float64 and not decimal
 CREATE INFIX OPERATOR
-std::`^` (n: std::float64, p: std::float64) -> std::float64
-    FROM SQL OPERATOR '^';
+std::`^` (n: std::int64, p: std::int64) -> std::float64
+{
+    FROM SQL OPERATOR r'*';
+    FROM SQL 'SELECT (n ^ p)';
```

Comment:
Shouldn't you be casing `p` to `float8` here to achieve the desired effect?

Reply:
It is my understanding that at the level of SQL operators we don't want to have unnecessary casts. As far as I can tell (albeit based on the signature of `pow` and my limited tests with `maxint64 ^ some_integer_power`) PostgreSQL actually returns `float8` for two integer inputs.

A possible test of the underlying implementation might be `10^309` which is greater than maxfloat64 and should result in an overflow error, while `10^308` is still a `float64`. If this test sounds reasonable, I can add it.

I can add the above comment about PostgreSQL to that specific definition, too.